Mathlib/LinearAlgebra/Dimension/Finite.lean

```diff
@@ -211,6 +211,54 @@ alias finset_card_le_finrank_of_linearIndependent := LinearIndependent.finset_ca
 @[deprecated]
 alias Module.Finite.lt_aleph0_of_linearIndependent := LinearIndependent.lt_aleph0_of_finite
 
+lemma exists_set_linearIndependent_of_lt_rank {n : Cardinal} (hn : n < Module.rank R M) :
+    ∃ s : Set M, #s = n ∧ LinearIndependent R ((↑) : s → M) := by
+  obtain ⟨⟨s, hs⟩, hs'⟩ := exists_lt_of_lt_ciSup' (hn.trans_eq (Module.rank_def R M))
+  obtain ⟨t, ht, ht'⟩ := le_mk_iff_exists_subset.mp hs'.le
+  exact ⟨t, ht', .mono ht hs⟩
```

Comment:
Could you change the condition of this result to `[Semiring R] [AddCommMonoid M] [Module R M]`? Same as the following `exists_finset_linearIndependent_of_le_rank`.

Reply:
It's OK if you do it in later PR; I only need to use the ring version.